Makefile

```diff
@@ -16,10 +16,12 @@ ifdef config
   endif
 endif
 
+PONYC ?= ponyc
```

Comment:
what's the purpose of this? One of the requirements for corral is that it can be bootstrapped by itself so I don't think we need this change.

Reply:
I'm going to merge with this in, but I'd like to discuss what the thinking was here as it is unrelated to the PR as far as I can tell.

Reply:
I needed to use a ponyc that is not on the path to test this against the recent master with the ProcessExitStatus changes. It doesnt hurt to be able to configure which ponyc to use for compilation while defaulting to ponyc which is first on the path. But i can revert it too, if this turns out to be problematic.